### PR TITLE
feat!: move to typescript code generation

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**

### CODE REVIEW

**Comment by xiaozhenliu-gg5** [comment]
Thanks for the suggestion @bcoe! This will make the script cleaner, and I agree we can do that in a separate PR coz this will spread over all APIs we have converted.

File: src/v1beta/os_login_service_client.ts [hunk 1]
```diff
@@ -0,0 +1,898 @@
...
+    );
+    const protos = gaxGrpc.loadProto(
+      opts.fallback ? require('../../protos/protos.json') : nodejsProtoPath
+    );
+
+    // This API contains "path templates"; forward-slash-separated
+    // identifiers to uniquely identify resources within the API.
+    // Create useful helper objects for these.
+    this._pathTemplates = {
+      posixAccountPathTemplate: new gaxModule.PathTemplate(
```
Comment (alexander-fenster) [inline src/v1beta/os_login_service_client.ts]: `fingerprintPathTemplate` is lost :(  Need to figure out what's going on with it.

**Comment by xiaozhenliu-gg5** [comment]
Yes sure, will take a look but no release attempt, thanks @alexander-fenster

File: src/v1beta/os_login_service_client.ts (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply src/v1beta/os_login_service_client.ts]: in gapic yaml `collections`, it does have resource defined as 
```
  - name_pattern: users/{user}/**sshPublicKeys**/{fingerprint}
    entity_name: fingerprint
```
while in the protos, this resource is not directly used, but referenced as 
```
    (google.api.resource_reference) = {
      type: "oslogin.googleapis.com/**SshPublicKey**"
    }
 OR
   string parent = 1 [(google.api.resource_reference) = {
                       child_type: "oslogin.googleapis.com/**SshPublicKey**"
                     }];
```
so we have `sshPublicKeyPathTemplate` but no `fingerprintPathTemplate`

Also one question is that as stated in [API conversion sheet](https://docs.google.com/spreadsheets/d/1Hyn_uPH1bzCK-9JAeUz_bj7feUg41h5hcuPzZRqUj8Q/edit#gid=0) this API use undefined resource `posixAccount` in ts file, we have `posixAccountPathTemplate` which is referenced to `project`:
```
      posixAccountPathTemplate: new gaxModule.PathTemplate(
        'users/{user}/projects/{project}'
      ),
```
Not sure if this is qualified, but all tests passed until now

Comment (alexander-fenster) [reply src/v1beta/os_login_service_client.ts]: Okay, thank you for investigation. I feel we might want to release it with the changed resource name, but it must be a major release (a breaking change). Let's merge it as `feat!` and not release yet (until we have a strategy about Node.js v8 deprecation).

**Pull request merged**